src/main/java/net/javadiscord/javabot/systems/starboard/StarboardManager.java

```diff
@@ -23,218 +31,169 @@
+		handleReactionEvent(event.getGuild(), event.getEmoji(), event.getChannel(), event.getMessageIdLong());
 	}
 
-	private void handleReactionEvent(Guild guild, MessageReaction.ReactionEmote reactionEmote, MessageChannel channel, long messageId) {
+	private void handleReactionEvent(Guild guild, Emoji emoji, MessageChannel channel, long messageId) {
 		Bot.asyncPool.submit(() -> {
-			var config = Bot.config.get(guild).getStarBoard();
+			StarboardConfig config = Bot.config.get(guild).getStarboardConfig();
 			if (config.getStarboardChannel().equals(channel)) return;
-			var starEmote = config.getEmotes().get(0);
-			if (!reactionEmote.getName().equals(starEmote)) return;
+			Emoji starEmote = config.getEmojis().get(0);
+			if (!emoji.equals(starEmote)) return;
 			channel.retrieveMessageById(messageId).queue(
 					message -> {
 						int stars = getReactionCountForEmote(starEmote, message);
-						try (var con = Bot.dataSource.getConnection()) {
-							var repo = new StarboardRepository(con);
-							var entry = repo.getEntryByMessageId(message.getIdLong());
+						DbHelper.doDaoAction(StarboardRepository::new, dao -> {
+							StarboardEntry entry = dao.getEntryByMessageId(message.getIdLong());
 							if (entry != null) {
 								updateStarboardMessage(message, stars, config);
 							} else if (stars >= config.getReactionThreshold()) {
 								addMessageToStarboard(message, stars, config);
-							} else if (stars < 1) {
-								if (!removeMessageFromStarboard(message.getIdLong(), channel, config)) {
-									log.error("Could not remove Message from Starboard");
-								}
+							} else if (stars < 1 && !removeMessageFromStarboard(message.getIdLong(), channel, config)) {
+								log.error("Could not remove Message from Starboard");
 							}
-						} catch (SQLException e) {
-							e.printStackTrace();
-						}
+						});
 					}, e -> log.error("Could not add Message to Starboard", e)
 			);
 		});
 	}
 
-	private boolean isValidChannel(MessageChannel channel) {
-		var type = channel.getType();
-		return type == ChannelType.TEXT || type == ChannelType.GUILD_PUBLIC_THREAD;
+	private boolean isInvalidChannel(@NotNull MessageChannel channel) {
+		ChannelType type = channel.getType();
+		return type != ChannelType.TEXT && type != ChannelType.GUILD_PUBLIC_THREAD;
 	}
 
 	@Override
 	public void onMessageDelete(@NotNull MessageDeleteEvent event) {
-		if (!isValidChannel(event.getChannel())) return;
-		try (var con = Bot.dataSource.getConnection()) {
-			var repo = new StarboardRepository(con);
-			var config = Bot.config.get(event.getGuild()).getStarBoard();
+		if (isInvalidChannel(event.getChannel())) return;
+		try (Connection con = Bot.dataSource.getConnection()) {
+			StarboardRepository repo = new StarboardRepository(con);
+			StarboardConfig config = Bot.config.get(event.getGuild()).getStarboardConfig();
 			StarboardEntry entry;
 			if (event.getChannel().equals(config.getStarboardChannel())) {
```

Comment:
`event.getChannel()` returns a `MessageChannelUnion` while `config.getStarboardChannel()` returns a `TextChannel`. These are not the same. Better compare the channel ids.